File: kotlin/bob/src/main/kotlin/Bob.kt
/**
 * Created by yannickgrenzinger on 16/08/2016.
 */
object Bob {

    fun hey(sentence: String): String {
        return when {
            sentence.matches(Regex("\\s*")) -> "Fine. Be that way!"
            sentence.matches(Regex("[^a-z-A-Z]+(?<!\\?)$")) -> "Whatever."
            sentence.matches(Regex("[^a-z-A-Z]+?$")) -> "Sure."
            sentence.matches(Regex("^[^\\p{Ll}]+$")) -> "Whoa, chill out!"
            sentence.matches(Regex("^(.*)\\?$")) -> "Sure."
            else -> "Whatever."
        }
    }


}

File: clojure/bob/src/bob.clj
 (ns bob)

 (defn response-for [sentence]
   (condp re-matches sentence
     #"\s*" "Fine. Be that way!"
     #"[^a-z-A-Z]+(?<!\?)$" "Whatever."
     #"[^a-z-A-Z]+?$" "Sure."
     #"^[^a-z]+$" "Whoa, chill out!"
     #"^(.*)\?$" "Sure."
     "Whatever."))
